### PR TITLE
Update changelog for v8.0.6

### PR DESCRIPTION
**Pull request by Gudahtt**
The update of `content-hash` was included because I wasn't entirely sure whether or not that involved a functional change, or what that change might be. I also neglected to expand upon the changes made in the `inpage-provider` update, as I couldn't think of a concise way to describe them.

### CODE REVIEW

**Review by whymarrh — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,13 @@
 ## Current Develop Branch
 
 ## 8.0.6 Wed Jul 22 2020
+- [#9030](https://github.com/MetaMask/metamask-extension/pull/9030): Hide "delete" button when editing contact of wallet account
+- [#9031](https://github.com/MetaMask/metamask-extension/pull/9031): Fix crash upon removing contact
+- [#9032](https://github.com/MetaMask/metamask-extension/pull/9032): Do not show spend limit for approvals
+- [#9046](https://github.com/MetaMask/metamask-extension/pull/9046): Update @metamask/inpage-provider@6.1.0
+- [#9048](https://github.com/MetaMask/metamask-extension/pull/9048): Skip attempts to resolve 0x contract prefix
+- [#9051](https://github.com/MetaMask/metamask-extension/pull/9051): Use content-hash@2.5.2
+- [#9056](https://github.com/MetaMask/metamask-extension/pull/9056): Display three significant digits for token balances
```
Comment (Gudahtt) [inline CHANGELOG.md:12]: Oh, upon reviewing this change again, this is an inaccurate way to describe this. I'll try again.

Comment (Gudahtt) [reply CHANGELOG.md:12]: Maybe this? cc @brad-decker 
```suggestion
- [#9056](https://github.com/MetaMask/metamask-extension/pull/9056): Display at least one significant digit of small non-zero token balances
```

Comment (Gudahtt) [reply CHANGELOG.md:12]: I'll go with this for now anyway

Comment (brad-decker) [reply CHANGELOG.md]: I think the suggestion you wrote works well @Gudahtt

**Review by brad-decker — APPROVED** [review]

**Pull request merged**